### PR TITLE
Allow `Explorer.Series.pow/2` on (number, lazy series)

### PR DESCRIPTION
**Pull request by nallwhy**
This PR solves #379

### CODE REVIEW

File: test/explorer/data_frame_test.exs [hunk 1]
```diff
@@ -560,7 +560,7 @@ defmodule Explorer.DataFrameTest do
                "calc2" => :integer,
                "calc3" => :integer,
                "calc4" => :float,
-               #  "calc5" => :integer,
+               "calc5" => :integer,
```
Comment (josevalim) [inline test/explorer/data_frame_test.exs:563]: Is this correct? The results above are float, no?

Comment (nallwhy) [reply test/explorer/data_frame_test.exs:563]: It's a bug #374.

dtype of series is float, but dtype of dataframe column is integer.

Comment (josevalim) [reply test/explorer/data_frame_test.exs:563]: So maybe we should leave this one commented and add a test where the first argument is a float?

**Review by josevalim — APPROVED** [review]

File: lib/explorer/series.ex [hunk 2]
```diff
@@ -1429,8 +1429,8 @@ defmodule Explorer.Series do
       >
   """
   @doc type: :element_wise
-  @spec pow(series :: Series.t(), right :: Series.t() | number()) :: Series.t()
-  def pow(%Series{} = series, right), do: basic_numeric_operation(:pow, series, right)
+  @spec pow(left :: Series.t() | number(), right :: Series.t() | number()) :: Series.t()
+  def pow(left, right), do: basic_numeric_operation(:pow, left, right)
```
Comment (philss) [inline lib/explorer/series.ex:1433]: I think we can merge this, but for what I understand the eager version of `Series.pow/2` is not going to work with numbers on the left-hand side.

**Review by philss — APPROVED** [review]

**Comment by philss** [comment]
@nallwhy thank you! :purple_heart:

**Pull request merged**